servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/LoadBalancerReadySubscriber.java

```diff
@@ -58,31 +57,29 @@ public void onNext(final Object o) {
         if (o instanceof LoadBalancerReadyEvent) {
             LoadBalancerReadyEvent event = (LoadBalancerReadyEvent) o;
             if (event.isReady()) {
-                Processor onHostsAvailable = LoadBalancerReadySubscriber.this.onHostsAvailable;
+                Processor onHostsAvailable = this.onHostsAvailable;
                 if (onHostsAvailable != null) {
-                    LoadBalancerReadySubscriber.this.onHostsAvailable = null;
+                    this.onHostsAvailable = null;
                     onHostsAvailable.onComplete();
                 }
-            } else if (LoadBalancerReadySubscriber.this.onHostsAvailable == null) {
-                LoadBalancerReadySubscriber.this.onHostsAvailable = newCompletableProcessor();
+            } else if (onHostsAvailable == null) {
+                onHostsAvailable = newCompletableProcessor();
             }
         }
     }
 
     @Override
     public void onError(final Throwable t) {
-        Processor onHostsAvailable = LoadBalancerReadySubscriber.this.onHostsAvailable;
+        Processor onHostsAvailable = this.onHostsAvailable;
         if (onHostsAvailable != null) {
-            LoadBalancerReadySubscriber.this.onHostsAvailable = null;
```

Comment:
Good catch!
Would be great to have a test for that

Reply:
tests added